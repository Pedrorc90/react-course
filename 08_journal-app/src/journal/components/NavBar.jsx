import { LogoutOutlined, MenuOutlined } from "@mui/icons-material";
import { AppBar, IconButton, Toolbar, Grid, Typography } from "@mui/material";
import { useDispatch } from "react-redux";
import { startLogout } from "../../store/auth/thunks";

const NavBar = ( { drawerWidth } ) => {

    const dispatch = useDispatch();

    const onLogout = () => {
        dispatch( startLogout() )
    }
    return ( 
        <AppBar position="fixed" sx={{ width: { sm: `calc(100% - ${ drawerWidth }px)` }, ml: { sm: `${ drawerWidth }px` } }}>
            <Toolbar>
                <IconButton color='inherit' edge="start" sx={{ mr: 2, display: { sm: 'none' } }}>
                    <MenuOutlined />
                </IconButton>

                <Grid container direction="row" justifyContent='space-around' alignItems="center">
                    <Typography variant="h6" noWrap component='div'>Journal App</Typography>
                    <IconButton color='error' onClick={ onLogout }>
                        <LogoutOutlined />
                    </IconButton>
                </Grid>
            </Toolbar>

        </AppBar>
     );
}
 
export default NavBar;