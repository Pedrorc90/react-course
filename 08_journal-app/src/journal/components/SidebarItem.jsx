import { TurnedInNot } from "@mui/icons-material"
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useMemo } from "react"
import { setActiveNote } from "../../store/journal/journalSlice"
import { useDispatch } from "react-redux"

export const SidebarItem = ( { title, body, id, date, imageUrls } ) =>{


    const newTitle = useMemo( () => {
        return title.length > 17 ? title.substring(0,17) + '...' : title
    }, [ title ] )

    const dispatch = useDispatch();

    const handleActiveNote = ({ title, body, id, date, imageUrls = [] }) => {
        const activeNote = { title, body, id, date, imageUrls };
        dispatch( setActiveNote( activeNote ) )
    }
    
    return (
        <ListItem key={ id } disablePadding>
            <ListItemButton onClick={ () => handleActiveNote({ title, body, id, date, imageUrls }) }>
                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                    <ListItemText primary={ newTitle } />
                    <ListItemText secondary={ body } />
                </Grid>
            </ListItemButton>
        </ListItem>
    )
      
}