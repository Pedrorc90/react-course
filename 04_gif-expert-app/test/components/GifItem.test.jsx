import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('GifItem component', () => {

    const title = 'Dragon Ball';
    const url = 'https://';

    test('Snapshot should be correct', () => {
        const { container } = render( <GifItem title={ title } url={ url }/> )
        expect( container ).toMatchSnapshot()
    })

    test('Should show the image with the url and the alt value', () => {
        render( <GifItem title={ title } url={ url }/> )
        expect( screen.getByRole('img').src).toBe( url )
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Should show the title in the component', () => {
        render( <GifItem title={ title } url={ url }/> )
        expect( screen.findByText( title ) ).toBeTruthy()
    })
})