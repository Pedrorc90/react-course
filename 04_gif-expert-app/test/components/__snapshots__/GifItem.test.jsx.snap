// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component Snapshot should be correct 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
